service/initialise_test.go

```diff
@@ -0,0 +1,188 @@
+package service
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-upload-service/api"
+	"github.com/ONSdigital/dp-upload-service/config"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+var err = func() error {
+	return errors.New("Server error")
+}
+
+var cfg, _ = config.Get()
+
+func TestGetHTTPServer(t *testing.T) {
+	Convey("Given a service list returns a server", t, func() {
+		serverMock := &HTTPServerMock{}
+		newServiceMock := &InitialiserMock{
+			DoGetHTTPServerFunc: func(bindAddr string, router http.Handler) HTTPServer {
+				return serverMock
+			},
+		}
+		r := mux.NewRouter()
+		svcList := NewServiceList(newServiceMock)
+		server := svcList.GetHTTPServer(cfg.BindAddr, r)
+		So(len(newServiceMock.DoGetHTTPServerCalls()), ShouldEqual, 1)
```

Comment:
continuation of previous comment
```suggestion
    Convey("Then the call has reached the right parts of the mock",...
		So(len(newServiceMock.DoGetHTTPServerCalls()), ShouldEqual, 1)
```

Reply:
would like to see this pattern repeated for each test (sorry!)